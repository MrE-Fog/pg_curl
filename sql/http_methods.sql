\unset ECHO
\set QUIET 1
\pset format unaligned
\pset tuples_only true
\pset pager off
\set ON_ERROR_ROLLBACK 1
\set ON_ERROR_STOP true
BEGIN;
CREATE EXTENSION pg_curl;
select curl_easy_reset();
select curl_easy_setopt_url('https://httpbin.org/get?a=b&c=&d');
select curl_easy_perform();
with s as (
    select regexp_matches(curl_easy_getinfo_header_in(), E'([^ \t\r\n\f]+): ?([^\t\r\n\f]+)', 'g') as s
) select s[1] as key, s[2] as value from s where s[1] not in ('date');
select  jsonb_pretty(jsonb_set_lax(convert_from(curl_easy_getinfo_data_in(), 'utf-8')::jsonb, '{headers,X-Amzn-Trace-Id}', null, true, 'delete_key'));

select curl_easy_reset();
select curl_easy_setopt_postfields(convert_to('{"e":"f","g":"","h":null}', 'utf-8'));
select curl_easy_setopt_url('https://httpbin.org/post?a=b&c=&d');
select curl_header_append('Content-Type', 'application/json; charset=utf-8');
select curl_easy_perform();
with s as (
    select regexp_matches(curl_easy_getinfo_header_in(), E'([^ \t\r\n\f]+): ?([^\t\r\n\f]+)', 'g') as s
) select s[1] as key, s[2] as value from s where s[1] not in ('date');
select  jsonb_pretty(jsonb_set_lax(convert_from(curl_easy_getinfo_data_in(), 'utf-8')::jsonb, '{headers,X-Amzn-Trace-Id}', null, true, 'delete_key'));
ROLLBACK;
