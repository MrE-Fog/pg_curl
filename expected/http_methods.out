\unset ECHO

t
t
content-type|application/json
content-length|274
access-control-allow-origin|*
access-control-allow-credentials|true
{
    "url": "https://httpbin.org/get?a=b&c=&d",
    "args": {
        "a": "b",
        "c": "",
        "d": ""
    },
    "headers": {
        "Host": "httpbin.org",
        "Accept": "*/*"
    }
}

t
t
t
t
content-type|application/json
content-length|502
access-control-allow-origin|*
access-control-allow-credentials|true
{
    "url": "https://httpbin.org/post?a=b&c=&d",
    "args": {
        "a": "b",
        "c": "",
        "d": ""
    },
    "data": "{\"e\":\"f\",\"g\":\"\",\"h\":null}",
    "form": {
    },
    "json": {
        "e": "f",
        "g": "",
        "h": null
    },
    "files": {
    },
    "headers": {
        "Host": "httpbin.org",
        "Accept": "*/*",
        "Content-Type": "application/json; charset=utf-8",
        "Content-Length": "25"
    }
}

t
t
t
content-type|application/json
content-length|468
access-control-allow-origin|*
access-control-allow-credentials|true
{
    "url": "https://httpbin.org/post?a=b&c=&d",
    "args": {
        "a": "b",
        "c": "",
        "d": ""
    },
    "data": "",
    "form": {
        "e": "f",
        "g": "",
        "h": ""
    },
    "json": null,
    "files": {
    },
    "headers": {
        "Host": "httpbin.org",
        "Accept": "*/*",
        "Content-Type": "application/x-www-form-urlencoded",
        "Content-Length": "8"
    }
}
